lesson02.md

```diff
@@ -0,0 +1,495 @@
+
+You can change the value of a variable using the assignment operator (`=`):
+
+```js
+let temperature = 24;   // define a variable, initialize it to 24
+temperature = 25;       // assign a new value (25) to the variable
+temperature = 23;       // now temperature is 23
+```
+
+---
+
+### Variables
+
+Variables can be used wherever we can use values:
+
+```js
+let pricePerTicket = 8;
+let friends = 3;
+let totalPrice = friends * pricePerTicket;
+```
+
+---
+
+### Why do we use variables?
+
+We use variables...
+
+* ... to give names (meaning) to values. `42` could mean a person’s age or shoe size.
+* ... to avoid repetitions. You define the variable once with a value, and use it often.
+
+---
+
+### Variable names
+
+You can name your variables in a lot of ways:
+
+* You can use letters, numbers, and underscore _ (spaces are *not* allowed!)
+* Variable name cannot start with a number.
+
+Correct variable names:
+```js
+let element = 3;
+let element3 = 5;
+```
+
+Wrong variable names:
+```js
+let 2squared = 4;
+let element-1 = 0;
+let full name = 'Anakin Skywalker';
+```
+
+---
+
+Giving good and descriptive names to your variables is very important!
+
+Good variable names make the code easier to understand by other developers, and even by yourself!
+
+Not wrong, but bad variable names:
+
+```js
+let a = 0;
+let _12 = 0;
+let asldjf = 0;
+let thisisareallylongvariablename = 0;
+```
+
+---
+
+In this course, and JavaScript in general, we use 'camelCase'
+  - `isStudent`
+  - `favoriteBook`
+  - `likesGermanFood`
+
+---
+
+Variable names are case sensitive:
+```js
+let name = 'John';
+let Name = 'John';
+let NAME = 'John';
+```
+
+These are 3 different variables.
+
+---
+
+### naming Boolean variables
+
+When naming a variables that points to a boolean, we can give it a name that describes a yes/no question:
+
+- `isMarried`
+- `hasKids`
+- `canExecute`
+
+---
+
+### try it!
+
+Create some variables:
+
+* one for your full name
+* one for your age
+* one for your favorite city
+* one for whether you can speak German or not
+* and anything else you can think of :)
+
+In the JavaScript console, enter the name of the variable. Do you see its value?
+
+Bonus: Change the value of one of the variables that you defined.
+
+---
+
+# Logical Operators
+
+---
+
+### Logical Operators
+
+These operators work on `boolean` values:
+
+* `&&` AND
+* `||` OR
+* `! ` NOT
+
+---
+
+### Logical NOT !
+
+Logical NOT negates a boolean:
+
+```javascript
+!true  // false
+!false // true
+```
+
+---
+
+Example: Carlo's Pizzeria doesn't serve pizza hawaii.
+
+```js
+let likesPizzaHawaii = true;
+let canVisitCarlo = !likesPizzaHawaii;
+```
+
+---
+
+### Quiz - Do we go biking or swimming?
+
+```js
+let temperature = 28;
+let isWarm = temperature > 25;
+let goSwimming = isWarm;
+let goBiking = !isWarm;
+```
+
+---
+
+### Try to prevent double negations
+
+```js
+let isNotWarm = temperature <= 25;
+let goSwimming = !isNotWarm; // No. Just no. Please Don't.
+```
```

Comment:
We could either ask learners or show in the next slide, how to avoid having double negation.

let isWarmEnough = temperature > 25;
let goSwimming = isWarmEnough;